### PR TITLE
Upgrade test code to latest versions, standards.

### LINKED ISSUE #513: Modernize tests.
I keep meaning to do this but I'm always in the middle of another change. Documenting as a reminder.

* Update Truth, JUnit, and compile-testing
* Change `ASSERT.about` to `assertAbout`
* Remove `Joiner` use for test sources, use string concatenation.

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #513.

### CODE REVIEW

File: compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java [hunk 1]
```diff
@@ -16,119 +16,128 @@
...
-            "  public Basic$Foo$Bar$$InjectAdapter() {",
-            "    super(\"Basic$Foo$Bar\", \"members/Basic$Foo$Bar\",",
-            "        NOT_SINGLETON, Basic.Foo$Bar.class);",
-            "  }",
-            "  @Override public Basic.Foo$Bar get() {",
-            "    Basic.Foo$Bar result = new Basic.Foo$Bar();",
-            "    return result;",
-            "  }",
-            "}"));
+        JavaFileObjects.forSourceString("Basic$Foo$Bar$$InjectAdapter", ""
```
Comment (swankjesse) [inline compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java]: no `\n` ?

File: compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java [hunk 2]
```diff
@@ -16,119 +16,128 @@
...
-            "    super(\"Basic$Foo$Bar$Baz\", \"members/Basic$Foo$Bar$Baz\",",
-            "        NOT_SINGLETON, Basic.Foo$Bar.Baz.class);",
-            "  }",
-            "  @Override public Basic.Foo$Bar.Baz get() {",
-            "    Basic.Foo$Bar.Baz result = new Basic.Foo$Bar.Baz();",
-            "    return result;",
-            "  }",
-            "}"));
+        JavaFileObjects.forSourceString("Basic$Foo$Bar$Baz$$InjectAdapter", ""
+            + "import dagger.internal.Binding;"
```
Comment (swankjesse) [inline compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java]: `\n` ?

File: compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java [hunk 3]
```diff
@@ -16,119 +16,128 @@
...
-            "  public Basic$A$$InjectAdapter() {",
-            "    super(\"Basic$A\", \"members/Basic$A\", NOT_SINGLETON, Basic.A.class);",
-            "  }",
-            "  @Override public Basic.A get() {",
-            "    Basic.A result = new Basic.A();",
-            "    return result;",
-            "  }",
-            "}"));
+        JavaFileObjects.forSourceString("Basic$A$$InjectAdapter", ""
+            + "import dagger.internal.Binding;"
```
Comment (swankjesse) [inline compiler/src/test/java/dagger/tests/integration/codegen/InjectAdapterGenerationTest.java]: `\n` ?

**Comment by swankjesse** [comment]
LGTM

**Comment by JakeWharton** [comment]
Added missing newlines.

**Pull request merged**